src/shared/sam/debugger/awsSamDebugger.ts

```diff
@@ -4,20 +4,44 @@
  */
 
 import * as vscode from 'vscode'
+import { CloudFormation } from '../../cloudformation/cloudformation'
+import { CloudFormationTemplateRegistry } from '../../cloudformation/templateRegistry'
 import { AwsSamDebuggerConfiguration } from './awsSamDebugConfiguration'
 
 export const AWS_SAM_DEBUG_TYPE = 'aws-sam'
+const DIRECT_INVOKE_TYPE = 'direct-invoke'
+const TEMPLATE_TARGET_TYPE = 'template'
+
 const AWS_SAM_DEBUG_REQUEST_TYPES = new Set<string>(['direct-invoke'])
 const AWS_SAM_DEBUG_TARGET_TYPES = new Set<string>(['template', 'code'])
 
 export class AwsSamDebugConfigurationProvider implements vscode.DebugConfigurationProvider {
-    public constructor() {}
+    public constructor(private readonly cftRegistry = CloudFormationTemplateRegistry.getRegistry()) {}
 
     public async provideDebugConfigurations(
         folder: vscode.WorkspaceFolder | undefined,
         token?: vscode.CancellationToken
     ): Promise<AwsSamDebuggerConfiguration[] | undefined> {
-        return undefined
+        if (folder) {
+            const debugConfigurations: AwsSamDebuggerConfiguration[] = []
+            const folderPath = folder.uri.fsPath
+            const templates = this.cftRegistry.registeredTemplates
+
+            for (const templateDatum of templates) {
+                if (templateDatum.path.startsWith(folderPath) && templateDatum.template.Resources) {
```

Comment:
can you double check that startswith isn't making a false positive match on a scenario where
workspace : `c:\foo`
template path : `c:\foobar\template.yaml`

Reply:
Hmm, will do.

Reply:
Done and has a test now.